pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -42,19 +42,22 @@ import (
 	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
 	"github.com/openshift/origin/pkg/dockerregistry/server"
 	"github.com/openshift/origin/pkg/dockerregistry/server/audit"
+	registryconfig "github.com/openshift/origin/pkg/dockerregistry/server/configuration"
+	"github.com/openshift/origin/pkg/dockerregistry/server/metrics"
 )
 
 // Execute runs the Docker registry.
 func Execute(configFile io.Reader) {
-	config, err := configuration.Parse(configFile)
+	dockerConfig, extraConfig, err := registryconfig.Parse(configFile)
 	if err != nil {
 		log.Fatalf("error parsing configuration file: %s", err)
 	}
-	setDefaultMiddleware(config)
-	setDefaultLogParameters(config)
+	setDefaultMiddleware(dockerConfig)
+	setDefaultLogParameters(dockerConfig)
 
 	ctx := context.Background()
-	ctx, err = configureLogging(ctx, config)
+	ctx = server.WithConfiguration(ctx, extraConfig)
```

Comment:
> I don't want to adв internal options to repository middleware.

@legionus It's fine to use the config object in `repository` middleware since we don't have a better way to pass the config there. On the other hand, there's no sense to have auth module differentiate between parameters passed via configuration file, parameters loaded from environment variables and parameters created on the fly. Those are assumptions that may easily become obsolete. Moreover, it makes unit tests hard to follow.

I still think we should pre-process the config and set `server.AccessControllerParams` accordingly. And do similar for all the other wrappers but `repository`. I won't block the review because of this though.
@dmage What is your opinion?